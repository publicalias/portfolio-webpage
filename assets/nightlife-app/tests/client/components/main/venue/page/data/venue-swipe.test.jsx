"use strict";

//local imports

const VenueSwipe = require("../../../../../../../scripts/client/components/main/venue/page/data/venue-swipe");

const { testWrapper } = require("../../../../../test-helpers");

//global imports

const { initTestEvent, initTestSnapshot, withDataList } = require("redux/tests/client-tests");
const { testMock } = require("redux/tests/meta-tests");
const { reactTests } = require("redux/tests/react-tests");

//utilities

const { testShallow } = testWrapper(VenueSwipe);

//setup

beforeAll(reactTests.setup);
beforeEach(reactTests.inject(VenueSwipe));

//venue swipe

describe("VenueSwipe", () => {

  const testSwipe = withDataList(testShallow, [null, {
    handleClick: jest.fn(),
    type: "meta"
  }]);

  const testSnapshot = initTestSnapshot(testSwipe);

  const testClick = initTestEvent(testSwipe, "click", {});

  it("should match snapshot", () => testSnapshot());

  it("should call handleClick on click", () => {

    const handleClick = jest.fn();

    return testClick(".qa-click-swipe", [null, { handleClick }], () => {
      testMock(handleClick, [{}]);
    });

  });

});
