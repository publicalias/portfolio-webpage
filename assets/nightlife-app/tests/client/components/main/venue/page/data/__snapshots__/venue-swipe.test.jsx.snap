// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VenueSwipe should match snapshot 1`] = `
<button
  className="c-venue-photos__swipe--meta qa-carousel-swipe"
  onClick={[MockFunction]}
>
  <i
    className="fas fa-chevron-meta"
  />
</button>
`;
